ush/soca/prep_ocean_obs.py

```diff
@@ -0,0 +1,253 @@
+        self.window_begin = window_begin_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+        self.window_end = window_begin_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        self.config.conversion_list_file = 'conversion_list.yaml'
+        self.config.save_list_file = 'save_list.yaml'
+
+    @logit(logger)
+    def initialize(self):
+        """Method initialize for ocean obs prep task
+        Parameters:
+        ------------
+        None
+        Returns:
+        --------
+        None
+        """
+
+        logger.info("initialize")
+
+        cdate = self.runtime_config['cdate']
+        cdatestr = cdate.strftime('%Y%m%d%H')
+        RUN = self.runtime_config.RUN
+        cyc = self.runtime_config['cyc']
+
+        OBS_YAML = self.config['OBS_YAML']
+        observer_config = YAMLFile(OBS_YAML)
+
+        OBSPREP_YAML = self.config['OBSPREP_YAML']
+        if os.path.exists(OBSPREP_YAML):
+            obsprep_config = YAMLFile(OBSPREP_YAML)
+        else:
+            logger.critical(f"OBSPREP_YAML file {OBSPREP_YAML} does not exist")
+            raise FileNotFoundError
+
+        JSON_TMPL_DIR = self.config.JSON_TMPL_DIR
+        BUFR2IODA_PY_DIR = self.config.BUFR2IODA_PY_DIR
+
+        COMIN_OBS = self.config.COMIN_OBS
+        COMOUT_OBS = self.config['COMOUT_OBS']
+        if not os.path.exists(COMOUT_OBS):
+            os.makedirs(COMOUT_OBS)
+
+        obsspaces_to_convert = []
+
+        try:
+            # go through the sources in OBS_YAML
+            for observer in observer_config['observers']:
+                try:
+                    obs_space_name = observer['obs space']['name']
+                    logger.info(f"Trying to find observation {obs_space_name} in OBSPREP_YAML")
+                except KeyError:
+                    logger.warning("Ill-formed observer yaml file, skipping")
+                    continue
+
+                # find match to the obs space from OBS_YAML in OBSPREP_YAML
+                # this is awkward and unpythonic, so feel free to improve
+                for obsprep_entry in obsprep_config['observations']:
+                    obsprep_space = obsprep_entry['obs space']
+                    obsprep_space_name = obsprep_space['name']
+
+                    if obsprep_space_name == obs_space_name:
+                        obtype = obsprep_space_name  # for brevity
+                        logger.info(f"Observer {obtype} found in OBSPREP_YAML")
+
+                        try:
+                            obs_window_back = obsprep_space['window']['back']
+                            obs_window_forward = obsprep_space['window']['forward']
+                        except KeyError:
+                            obs_window_back = 0
+                            obs_window_forward = 0
+
+                        window_cdates = []
+                        for i in range(-obs_window_back, obs_window_forward + 1):
+                            interval = timedelta(hours=6 * i)
```

Comment:
same comment here about the window length

Reply:
done